src/Aspire.Hosting/Dcp/DcpHostService.cs

```diff
@@ -73,7 +74,12 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             return;
         }
 
-        EnsureDockerIfNecessary();
+        var dockerHealthCheckTask = EnsureDockerIfNecessaryAsync(cancellationToken);
+        var dcpVersionCheckTask = EnsureDcpVersionAsync(cancellationToken);
+
+        await dockerHealthCheckTask.ConfigureAwait(false);
```

Comment:
Isn't this already parallel, since the tasks are both created (and thus launched) before either is awaited? Or am I misunderstanding/misremembering .NET's async/await model?

Reply:
You are absolutely right. I am just so used to the fact that in a typical .NET code `await` is paired with a call that starts a task, that I did not see it was not the case here. But I like your change; `Task.WhenAll()` makes it obvious.